kratos/containers/csr_matrix.h

```diff
@@ -593,8 +592,45 @@ class CsrMatrix final
     // RightScaling
     // SymmetricScaling
 
-    //TODO
-    //void ApplyDirichlet
+    template<class TVectorType1, class TVectorType2=TVectorType1>
+    void ApplyHomogeneousDirichlet(const TVectorType1& free_dofs_vector, 
+                                    const TDataType DiagonalValue, 
+                                    TVectorType2& rRHS)
+    {
+        KRATOS_ERROR_IF(size1() != free_dofs_vector.size() ) << "ApplyDirichlet: mismatch between row sizes : " << size1()  
+            << " and free_dofs_vector size " << free_dofs_vector.size() << std::endl;
+        KRATOS_ERROR_IF(size2() != free_dofs_vector.size() ) << "ApplyDirichlet: mismatch between col sizes : " << size2()  
+            << " and free_dofs_vector size " << free_dofs_vector.size() << std::endl;
+
+        if(nnz() != 0)
```

Comment:
I just want to mention that our style is:

~~~
if () {
} else {
}
~~~

Idem for the others

Reply:
i think it is clearer as it is now.

I passed astyle on the file for other style correcitons